### PR TITLE
Add module PoolSNP

### LINKED ISSUE #5953: new module: PoolSNP
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by abhilesh**
This PR adds a module for [PoolSNP](https://github.com/capoony/PoolSNP), a heuristic SNP caller for pooled sequencing data. 

The module has takes in a `.mpileup` file and reference `.fasta` as inputs and outputs the following  
- `.vcf.gz`, a gzipped VCF file containing allele counts and frequencies for every position and library
- `*cov-*.txt`, a max coverage file containing the maximum coverage thresholds for all chromosomal arms and libraries in the mpileup file
- `*BS.txt.gz`, Optionally, a bad sites file, containing a list of (variable and invariable) sites that did not pass the SNP calling criteria

## PR checklist

Closes #5953 

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [X] `nf-core modules test <MODULE> --profile docker`
    - [X] `nf-core modules test <MODULE> --profile singularity`
    - [X] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/poolsnp/main.nf [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/poolsnp:1.0.1--py312h7e72e81_0':
+        'biocontainers/poolsnp:1.0.1--py312h7e72e81_0' }"
+
+    input:
+    tuple val(meta)      , path(mpileup)
+    tuple val(meta2)     , path(reference)
+    val(max_cov)
```
Comment (SPPearce) [inline modules/nf-core/poolsnp/main.nf:13]: I think this is the only required parameter with no default, so you don't need to expose the others as value channels.

Comment (abhilesh) [reply modules/nf-core/poolsnp/main.nf:13]: Removed the parameters with default values, also removed the corresponding entries from `meta.yml`, these parameters shouldn't be included in the `meta.yml` right?

Passing the other parameters through `ext.args` as we get an empty `.vcf` file with the default values.

Comment (SPPearce) [reply modules/nf-core/poolsnp/main.nf:13]: Ok, definitely sensible to pass the other parameters for the test to make an actual output

**Comment by SPPearce** [comment]
The tool documentation suggests for the `max-cov` option:
> This cutoff can either be newly calculated by providing the threshold percentile or by providing the full path to a max-coverage file.
So we should try and ensure that option is supported.

**Comment by abhilesh** [comment]
> The tool documentation suggests for the `max-cov` option:
> 
> > This cutoff can either be newly calculated by providing the threshold percentile or by providing the full path to a max-coverage file.
> > So we should try and ensure that option is supported.

Any suggestions on how to go about implementing this?

The parameter `max-cov` would need to be either a file type or an integer value, but not both. We can't specify optional inputs for nextflow processes so both the input channels need to be declared and then if a file is passed as the parameter, we use that channel instead of the value channel?

**Comment by abhilesh** [comment]
- Added the option to specify `max-cov` as either a float value or a tsv file.
- Assert the presence of only one type of input
- Added tests for both cases when `max-cov` is a value channel vs a path channel

**Review by SPPearce — COMMENTED** [review]
Can you add a stub test as well please.

**Review by SPPearce — APPROVED** [review]
Looks good to go now

**Pull request merged**